alembic/versions/ec5d2787bec9_external_links.py

```diff
@@ -0,0 +1,34 @@
+"""External links
+
+Revision ID: ec5d2787bec9
+Revises: 7a345f1bf9c3
+Create Date: 2024-05-29 06:39:17.930675
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+from sqlalchemy.orm import Session
+
+from mavedb.models.score_set import ScoreSet
+
+# revision identifiers, used by Alembic.
+revision = 'ec5d2787bec9'
+down_revision = '7a345f1bf9c3'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.add_column("scoresets", sa.Column("external_links", postgresql.JSONB(astext_type=sa.Text()), nullable=True))
```

Comment:
As an alternative to the below, I think you could do something like 
```
op.add_column("scoresets", sa.Column("external_links", postgresql.JSONB(astext_type=sa.Text()), nullable=False, server_default='{}'))
```
which will set the default of each of the rows as the column is added so you don't need to mess with the nullable property and access the DB. See https://docs.sqlalchemy.org/en/20/core/metadata.html#sqlalchemy.schema.Column.params.server_default

Reply:
Thanks, I like that better. I wonder why it can't generate that from the SQLAlchemy model with its default, but maybe JSONB is tricky or it doesn't want to assume that we want a server-side default.

Reply:
Yeah not sure-- the auto-generated migrations don't handle new non-nullable columns that well imo.